### PR TITLE
Convert Spaces to Underscores for Preset and Tag Names

### PR DESCRIPTION
**Pull request by aNOOBisTheGod**
Now, user can enter tag or preset name with spacebars and they will be changed to underlines(and also in database, so there shouldn't be any cringe spacebar bugs). I think that will be useful

### CODE REVIEW

**Comment by Miodec** [comment]
Build is failing - please fix

**Comment by aNOOBisTheGod** [comment]
> Build is failing - please fix

done

File: frontend/src/ts/db.ts [hunk 1]
```diff
@@ -125,6 +125,11 @@ export async function initSnapshot(): Promise<
     // LoadingPage.updateText("Downloading tags...");
     snap.customThemes = userData.customThemes ?? [];
     snap.tags = tagsData;
+
+    snap.tags.forEach((tag) => {
+      tag.display = tag.name.replaceAll("_", " ");
```
Comment (Bruception) [inline frontend/src/ts/db.ts:130]: I don’t understand, display vs name?

Comment (aNOOBisTheGod) [reply frontend/src/ts/db.ts:130]: display is what user will see on website, name is what will be in the database

Comment (Bruception) [reply frontend/src/ts/db.ts:130]: Just store it with underscores

File: frontend/src/ts/ape/endpoints/presets.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ export default function getPresetsEndpoints(
     configChanges: MonkeyTypes.ConfigChanges
   ): Ape.EndpointData {
     const payload = {
-      name: presetName,
+      name: presetName.replaceAll(" ", "_"),
```
Comment (Miodec) [inline frontend/src/ts/ape/endpoints/presets.ts:15]: No need to replace in Ape if you already replace it in the popup code right after the user inputs it. Please remove.

**Review by Miodec — CHANGES_REQUESTED** [review]

**Comment by Miodec** [comment]
Have you done any testing?

**Comment by aNOOBisTheGod** [comment]
> Have you done any testing?

Yep. It works for me

**Comment by Miodec** [comment]
Tags not using `display` on the typing page and results page.
image

**Comment by aNOOBisTheGod** [comment]
Done, guess now displays properly

File: frontend/src/ts/ape/endpoints/presets.ts (hunk 2 above)
Comment (aNOOBisTheGod) [reply frontend/src/ts/ape/endpoints/presets.ts]: removed

**Review by Miodec — APPROVED** [review]
Thanks

**Pull request merged**